Index: code/ui/ui_main.c
===================================================================
--- code/ui/ui_main.c	(revision 3844)
+++ code/ui/ui_main.c	(working copy)
@@ -3021,6 +3021,18 @@
 			trap_Cmd_ExecuteText( EXEC_APPEND, "disconnect\n" );
 			UI_EnterMenu();
 			Menus_CloseAll();
+
+			if (ui_singlePlayerActive.integer) {
+				// put back all the ui overrides
+				trap_Cvar_SetValue("capturelimit", trap_Cvar_VariableValue("ui_saveCaptureLimit"));
+				trap_Cvar_SetValue("fraglimit", trap_Cvar_VariableValue("ui_saveFragLimit"));
+				trap_Cvar_SetValue("cg_drawTimer", trap_Cvar_VariableValue("ui_drawTimer"));
+				trap_Cvar_SetValue("g_doWarmup", trap_Cvar_VariableValue("ui_doWarmup"));
+				trap_Cvar_SetValue("g_Warmup", trap_Cvar_VariableValue("ui_Warmup"));
+				trap_Cvar_SetValue("sv_pure", trap_Cvar_VariableValue("ui_pure"));
+				trap_Cvar_SetValue("g_friendlyFire", trap_Cvar_VariableValue("ui_friendlyFire"));
+			}
+
 			Menus_ActivateByName("main");
 		} else if (Q_stricmp(name, "ServerSort") == 0) {
 			int sortColumn;
