Index: code/qcommon/msg.c
===================================================================
--- code/qcommon/msg.c	(revision 2398)
+++ code/qcommon/msg.c	(working copy)
@@ -37,8 +37,6 @@
 ==============================================================================
 */
 
-int oldsize = 0;
-
 void MSG_initHuffman( void );
 
 void MSG_Init( msg_t *buf, byte *data, int length ) {
@@ -105,8 +103,6 @@
 void MSG_WriteBits( msg_t *msg, int value, int bits ) {
 	int	i;
 
-	oldsize += bits;
-
 	if ( msg->overflowed ) {
 		return;
 	}
@@ -653,7 +649,6 @@
 		from->buttons == to->buttons &&
 		from->weapon == to->weapon) {
 			MSG_WriteBits( msg, 0, 1 );				// no change
-			oldsize += 7;
 			return;
 	}
 	key ^= to->serverTime;
@@ -875,8 +870,8 @@
 
 	MSG_WriteByte( msg, lc );	// # of changes
 
-	oldsize += numFields;
 
 	for ( i = 0, field = entityStateFields ; i < lc ; i++, field++ ) {
 		fromF = (int *)( (byte *)from + field->offset );
 		toF = (int *)( (byte *)to + field->offset );
@@ -895,7 +890,6 @@
 
 			if (fullFloat == 0.0f) {
 					MSG_WriteBits( msg, 0, 1 );
-					oldsize += FLOAT_INT_BITS;
 			} else {
 				MSG_WriteBits( msg, 1, 1 );
 				if ( trunc == fullFloat && trunc + FLOAT_INT_BIAS >= 0 && 
@@ -1150,7 +1144,6 @@
 
 	MSG_WriteByte( msg, lc );	// # of changes
 
-	oldsize += numFields - lc;
 
 	for ( i = 0, field = playerStateFields ; i < lc ; i++, field++ ) {
 		fromF = (int *)( (byte *)from + field->offset );
@@ -1216,7 +1209,6 @@
 
 	if (!statsbits && !persistantbits && !ammobits && !powerupbits) {
 		MSG_WriteBits( msg, 0, 1 );	// no change
-		oldsize += 4;
 		return;
 	}
 	MSG_WriteBits( msg, 1, 1 );	// changed
