Index: code/client/snd_codec_ogg.c
===================================================================
--- code/client/snd_codec_ogg.c	(revision 104)
+++ code/client/snd_codec_ogg.c	(working copy)
@@ -129,7 +129,7 @@
 			retVal = FS_Seek(stream->file, (long) offset, FS_SEEK_SET);
 
 			// something has gone wrong, so we return here
-			if(retVal < 0)
+			if(retVal != 0)
 			{
 			 return retVal;
 			}
@@ -145,7 +145,7 @@
 			retVal = FS_Seek(stream->file, (long) offset, FS_SEEK_CUR);
 
 			// something has gone wrong, so we return here
-			if(retVal < 0)
+			if(retVal != 0)
 			{
 			 return retVal;
 			}
@@ -161,7 +161,7 @@
 			retVal = FS_Seek(stream->file, (long) offset, FS_SEEK_END);
 
 			// something has gone wrong, so we return here
-			if(retVal < 0)
+			if(retVal != 0)
 			{
 			 return retVal;
 			}
Index: code/client/snd_codec_opus.c
===================================================================
--- code/client/snd_codec_opus.c	(revision 104)
+++ code/client/snd_codec_opus.c	(working copy)
@@ -118,7 +118,7 @@
 			retVal = FS_Seek(stream->file, (long) offset, FS_SEEK_SET);
 
 			// something has gone wrong, so we return here
-			if(retVal < 0)
+			if(retVal != 0)
 			{
 			 return retVal;
 			}
@@ -134,7 +134,7 @@
 			retVal = FS_Seek(stream->file, (long) offset, FS_SEEK_CUR);
 
 			// something has gone wrong, so we return here
-			if(retVal < 0)
+			if(retVal != 0)
 			{
 			 return retVal;
 			}
@@ -150,7 +150,7 @@
 			retVal = FS_Seek(stream->file, (long) offset, FS_SEEK_END);
 
 			// something has gone wrong, so we return here
-			if(retVal < 0)
+			if(retVal != 0)
 			{
 			 return retVal;
 			}
Index: code/qcommon/files.c
===================================================================
--- code/qcommon/files.c	(revision 104)
+++ code/qcommon/files.c	(working copy)
@@ -1631,6 +1631,7 @@
 =================
 FS_Seek
 
+Returns 0 on success and non-zero on failure
 =================
 */
 int FS_Seek( fileHandle_t f, long offset, int origin ) {
@@ -1681,7 +1682,7 @@
 		switch( origin ) {
 			case FS_SEEK_SET:
 				if ( remainder == currentPosition ) {
-					return offset;
+					return 0;
 				}
 				unzSetOffset(fsh[f].handleFiles.file.z, fsh[f].zipFilePos);
 				unzOpenCurrentFile(fsh[f].handleFiles.file.z);
@@ -1694,7 +1695,7 @@
 					remainder -= PK3_SEEK_BUFFER_SIZE;
 				}
 				FS_Read( buffer, remainder, f );
-				return offset;
+				return 0;
 
 			default:
 				Com_Error( ERR_FATAL, "Bad origin in FS_Seek" );
