Index: code/game/g_team.c
===================================================================
--- code/game/g_team.c	(revision 3367)
+++ code/game/g_team.c	(working copy)
@@ -710,11 +710,11 @@
 		return 0; // We don't have the flag
 #ifdef MISSIONPACK
 	if( g_gametype.integer == GT_1FCTF ) {
-		PrintMsg( NULL, "%s" S_COLOR_WHITE " captured the flag!\n", cl->pers.netname );
+		trap_SendServerCommand( -1, va("cp \"%s" S_COLOR_WHITE "\ncaptured the flag!\n\"", cl->pers.netname ));
 	}
 	else {
 #endif
-	PrintMsg( NULL, "%s" S_COLOR_WHITE " captured the %s flag!\n", cl->pers.netname, TeamName(OtherTeam(team)));
+	trap_SendServerCommand( -1, va("cp \"%s" S_COLOR_WHITE "\ncaptured the %s flag!\n\"", cl->pers.netname, TeamName(OtherTeam(team))));
 #ifdef MISSIONPACK
 	}
 #endif
@@ -1242,19 +1242,26 @@
 	attacker->client->rewardTime = level.time + REWARD_SPRITE_TIME;
 	attacker->client->ps.persistant[PERS_CAPTURES]++;
 
+	Team_CaptureFlagSound(self, self->spawnflags);
+
 	teamgame.redObeliskAttackedTime = 0;
 	teamgame.blueObeliskAttackedTime = 0;
+
+	trap_SendServerCommand( -1, va("cp \"%s" S_COLOR_WHITE "\ndestroyed the %s obelisk.\n\"",
+			attacker->client->pers.netname, TeamName(self->spawnflags)));
 }
 
 
 static void ObeliskTouch( gentity_t *self, gentity_t *other, trace_t *trace ) {
 	int			tokens;
+	team_t		otherTeam;
 
 	if ( !other->client ) {
 		return;
 	}
 
-	if ( OtherTeam(other->client->sess.sessionTeam) != self->spawnflags ) {
+	otherTeam = OtherTeam( other->client->sess.sessionTeam );
+	if ( otherTeam != self->spawnflags ) {
 		return;
 	}
 
@@ -1263,8 +1270,8 @@
 		return;
 	}
 
-	PrintMsg(NULL, "%s" S_COLOR_WHITE " brought in %i %s.\n",
-					other->client->pers.netname, tokens, ( tokens == 1 ) ? "skull" : "skulls" );
+	trap_SendServerCommand( -1, va( "cp \"%s" S_COLOR_WHITE "\nbrought in %i %s %s.\n\"",
+					other->client->pers.netname, tokens, TeamName( otherTeam ), ( tokens == 1 ) ? "skull" : "skulls" ) );
 
 	AddTeamScore(self->s.pos.trBase, other->client->sess.sessionTeam, tokens);
 	Team_ForceGesture(other->client->sess.sessionTeam);
