Index: code/cgame/cg_local.h
===================================================================
--- code/cgame/cg_local.h	(revision 3371)
+++ code/cgame/cg_local.h	(working copy)
@@ -79,15 +79,6 @@
 #define TEAM_OVERLAY_MAXNAME_WIDTH	12
 #define TEAM_OVERLAY_MAXLOCATION_WIDTH	16
 
-#define	DEFAULT_MODEL			"sarge"
-#ifdef MISSIONPACK
-#define	DEFAULT_TEAM_MODEL		"james"
-#define	DEFAULT_TEAM_HEAD		"*james"
-#else
-#define	DEFAULT_TEAM_MODEL		"sarge"
-#define	DEFAULT_TEAM_HEAD		"sarge"
-#endif
-
 typedef enum {
 	FOOTSTEP_NORMAL,
 	FOOTSTEP_BOOT,
Index: code/cgame/cg_main.c
===================================================================
--- code/cgame/cg_main.c	(revision 3371)
+++ code/cgame/cg_main.c	(working copy)
@@ -353,7 +353,7 @@
 	blueTeamNameModificationCount = cg_blueTeamName.modificationCount;
 #endif
 	trap_Cvar_Register(NULL, "model", DEFAULT_MODEL, CVAR_USERINFO | CVAR_ARCHIVE );
-	trap_Cvar_Register(NULL, "headmodel", DEFAULT_MODEL, CVAR_USERINFO | CVAR_ARCHIVE );
+	trap_Cvar_Register(NULL, "headmodel", DEFAULT_HEAD, CVAR_USERINFO | CVAR_ARCHIVE );
 	trap_Cvar_Register(NULL, "team_model", DEFAULT_TEAM_MODEL, CVAR_USERINFO | CVAR_ARCHIVE );
 	trap_Cvar_Register(NULL, "team_headmodel", DEFAULT_TEAM_HEAD, CVAR_USERINFO | CVAR_ARCHIVE );
 }
Index: code/client/cl_main.c
===================================================================
--- code/client/cl_main.c	(revision 3371)
+++ code/client/cl_main.c	(working copy)
@@ -3618,13 +3618,13 @@
 	cl_consoleKeys = Cvar_Get( "cl_consoleKeys", "~ ` 0x7e 0x60", CVAR_ARCHIVE);
 
 	// userinfo
-	Cvar_Get ("name", "UnnamedPlayer", CVAR_USERINFO | CVAR_ARCHIVE );
+	Cvar_Get ("name", DEFAULT_PLAYER_NAME, CVAR_USERINFO | CVAR_ARCHIVE );
 	cl_rate = Cvar_Get ("rate", "25000", CVAR_USERINFO | CVAR_ARCHIVE );
 	Cvar_Get ("snaps", "20", CVAR_USERINFO | CVAR_ARCHIVE );
-	Cvar_Get ("model", "sarge", CVAR_USERINFO | CVAR_ARCHIVE );
-	Cvar_Get ("headmodel", "sarge", CVAR_USERINFO | CVAR_ARCHIVE );
-	Cvar_Get ("team_model", "james", CVAR_USERINFO | CVAR_ARCHIVE );
-	Cvar_Get ("team_headmodel", "*james", CVAR_USERINFO | CVAR_ARCHIVE );
+	Cvar_Get ("model", DEFAULT_MODEL, CVAR_USERINFO | CVAR_ARCHIVE );
+	Cvar_Get ("headmodel", DEFAULT_HEAD, CVAR_USERINFO | CVAR_ARCHIVE );
+	Cvar_Get ("team_model", DEFAULT_TEAM_MODEL, CVAR_USERINFO | CVAR_ARCHIVE );
+	Cvar_Get ("team_headmodel", DEFAULT_TEAM_HEAD, CVAR_USERINFO | CVAR_ARCHIVE );
 	Cvar_Get ("color1",  "4", CVAR_USERINFO | CVAR_ARCHIVE );
 	Cvar_Get ("color2", "5", CVAR_USERINFO | CVAR_ARCHIVE );
 	Cvar_Get ("handicap", "100", CVAR_USERINFO | CVAR_ARCHIVE );
Index: code/game/bg_public.h
===================================================================
--- code/game/bg_public.h	(revision 3371)
+++ code/game/bg_public.h	(working copy)
@@ -540,7 +540,14 @@
 // changes so a restart of the same anim can be detected
 #define	ANIM_TOGGLEBIT		128
 
+#define DEFAULT_PLAYER_NAME		"UnnamedPlayer"
+// Default player model names
+#define DEFAULT_MODEL			"sarge"
+#define DEFAULT_HEAD			"sarge"
+// Default team player model names
+#define DEFAULT_TEAM_MODEL		"james"
+#define DEFAULT_TEAM_HEAD		"*james"
 
+#define DEFAULT_REDTEAM_NAME		"Pagans"
+#define DEFAULT_BLUETEAM_NAME		"Stroggs"
 typedef enum {
 	TEAM_FREE,
 	TEAM_RED,
@@ -550,8 +557,6 @@
 	TEAM_NUM_TEAMS
 } team_t;
 
-#define DEFAULT_REDTEAM_NAME "Pagans"
-#define DEFAULT_BLUETEAM_NAME "Stroggs"
 // Time between location updates
 #define TEAM_LOCATION_UPDATE_TIME		1000
 
Index: code/game/g_bot.c
===================================================================
--- code/game/g_bot.c	(revision 3368)
+++ code/game/g_bot.c	(working copy)
@@ -591,6 +591,7 @@
 	char			*model;
 	char			*headmodel;
 	char			userinfo[MAX_INFO_STRING];
+	qboolean		modelSet;
 
 	// have the server allocate a client slot
 	clientNum = trap_BotAllocateClient();
@@ -679,8 +680,9 @@
 
 	key = "model";
 	model = Info_ValueForKey( botinfo, key );
-	if ( !*model ) {
-		model = "visor/default";
+	modelSet = ( *model );
+	if ( !modelSet ) {
+		model = DEFAULT_MODEL;
 	}
 	Info_SetValueForKey( userinfo, key, model );
 	key = "team_model";
@@ -689,7 +691,11 @@
 	key = "headmodel";
 	headmodel = Info_ValueForKey( botinfo, key );
 	if ( !*headmodel ) {
-		headmodel = model;
+		if ( !modelSet ) {
+			headmodel = DEFAULT_HEAD;
+		} else {
+			headmodel = model;
+		}
 	}
 	Info_SetValueForKey( userinfo, key, headmodel );
 	key = "team_headmodel";
@@ -819,7 +825,7 @@
 	for (i = 0; i < g_numBots; i++) {
 		Q_strncpyz(name, Info_ValueForKey( g_botInfos[i], "name" ), sizeof( name ));
 		if ( !*name ) {
-			strcpy(name, "UnnamedPlayer");
+			strcpy(name, DEFAULT_PLAYER_NAME);
 		}
 		Q_strncpyz(funname, Info_ValueForKey( g_botInfos[i], "funname" ), sizeof( funname ));
 		if ( !*funname ) {
@@ -827,7 +833,7 @@
 		}
 		Q_strncpyz(model, Info_ValueForKey( g_botInfos[i], "model" ), sizeof( model ));
 		if ( !*model ) {
-			strcpy(model, "visor/default");
+			strcpy(model, DEFAULT_MODEL);
 		}
 		Q_strncpyz(aifile, Info_ValueForKey( g_botInfos[i], "aifile"), sizeof( aifile ));
 		if (!*aifile ) {
Index: code/game/g_client.c
===================================================================
--- code/game/g_client.c	(revision 3371)
+++ code/game/g_client.c	(working copy)
@@ -685,7 +685,7 @@
 
 	// don't allow empty names
 	if( *out == '\0' || colorlessLen == 0)
-		Q_strncpyz(out, "UnnamedPlayer", outSize );
+		Q_strncpyz(out, DEFAULT_PLAYER_NAME, outSize );
 }
 
 
Index: code/ui/ui_main.c
===================================================================
--- code/ui/ui_main.c	(revision 3371)
+++ code/ui/ui_main.c	(working copy)
@@ -5866,7 +5866,7 @@
 	{ &ui_captureLimit, "ui_captureLimit", "5", 0},
 	{ &ui_smallFont, "ui_smallFont", "0.25", CVAR_ARCHIVE},
 	{ &ui_bigFont, "ui_bigFont", "0.4", CVAR_ARCHIVE},
-	{ &ui_findPlayer, "ui_findPlayer", "Sarge", CVAR_ARCHIVE},
+	{ &ui_findPlayer, "ui_findPlayer", DEFAULT_MODEL, CVAR_ARCHIVE},
 	{ &ui_Q3Model, "ui_q3model", "0", CVAR_ARCHIVE},
 	{ &ui_hudFiles, "cg_hudFiles", "ui/hud.txt", CVAR_ARCHIVE},
 	{ &ui_recordSPDemo, "ui_recordSPDemo", "0", CVAR_ARCHIVE},
