Index: code/botlib/be_ai_move.c
===================================================================
--- code/botlib/be_ai_move.c	(revision 3349)
+++ code/botlib/be_ai_move.c	(working copy)
@@ -1541,8 +1541,14 @@
 	//elemantary actions
 	//EA_Move(ms->client, dir, 400);
 	EA_MoveForward(ms->client);
-	//move up if close to the actual out of water jump spot
-	if (dist < 40) EA_MoveUp(ms->client);
+	//move up if close to the actual out of water jump spot
+	if (dist < 40) {
+		if (ms->moveflags & MFL_ONGROUND) {
+			EA_Jump(ms->client);
+		} else {
+			EA_MoveUp(ms->client);
+		}
+	}
 	//set the ideal view angles
 	vectoangles(dir, result.ideal_viewangles);
 	result.flags |= MOVERESULT_MOVEMENTVIEW;
