Index: code/botlib/be_aas_sample.c
===================================================================
--- code/botlib/be_aas_sample.c	(revision 3344)
+++ code/botlib/be_aas_sample.c	(working copy)
@@ -74,7 +74,7 @@
 	int index;
 	//bounding box size for each presence type
 	vec3_t boxmins[3] = {{0, 0, 0}, {-15, -15, -24}, {-15, -15, -24}};
-	vec3_t boxmaxs[3] = {{0, 0, 0}, { 15,  15,  56}, { 15,  15,   8}};
+	vec3_t boxmaxs[3] = {{0, 0, 0}, { 15,  15,  56}, { 15,  15, 42}}; // Tobias CHECK: why was this 8 and not 16 (see 'PM_CheckDuck' etc.)?
 
 	if (presencetype == PRESENCE_NORMAL) index = 1;
 	else if (presencetype == PRESENCE_CROUCH) index = 2;
Index: code/game/bg_pmove.c
===================================================================
--- code/game/bg_pmove.c	(revision 3344)
+++ code/game/bg_pmove.c	(working copy)
@@ -1303,7 +1303,7 @@
 
 	if (pm->ps->pm_flags & PMF_DUCKED)
 	{
-		pm->maxs[2] = 16;
+		pm->maxs[2] = 42;
 		pm->ps->viewheight = CROUCH_VIEWHEIGHT;
 	}
 	else
