Index: code/botlib/be_aas_move.c
===================================================================
--- code/botlib/be_aas_move.c	(revision 3347)
+++ code/botlib/be_aas_move.c	(working copy)
@@ -89,8 +89,8 @@
 	aassettings.phys_gravity				= LibVarValue("phys_gravity", "800");
 	aassettings.phys_waterfriction			= LibVarValue("phys_waterfriction", "1");
 	aassettings.phys_watergravity			= LibVarValue("phys_watergravity", "400");
-	aassettings.phys_maxvelocity			= LibVarValue("phys_maxvelocity", "320");
-	aassettings.phys_maxwalkvelocity		= LibVarValue("phys_maxwalkvelocity", "320");
+	aassettings.phys_maxvelocity			= LibVarValue("phys_maxvelocity", "260");
+	aassettings.phys_maxwalkvelocity		= LibVarValue("phys_maxwalkvelocity", "280");
 	aassettings.phys_maxcrouchvelocity		= LibVarValue("phys_maxcrouchvelocity", "100");
 	aassettings.phys_maxswimvelocity		= LibVarValue("phys_maxswimvelocity", "150");
 	aassettings.phys_walkaccelerate			= LibVarValue("phys_walkaccelerate", "10");
@@ -98,9 +98,9 @@
 	aassettings.phys_swimaccelerate			= LibVarValue("phys_swimaccelerate", "4");
 	aassettings.phys_maxstep				= LibVarValue("phys_maxstep", "19");
 	aassettings.phys_maxsteepness			= LibVarValue("phys_maxsteepness", "0.7");
-	aassettings.phys_maxwaterjump			= LibVarValue("phys_maxwaterjump", "18");
-	aassettings.phys_maxbarrier				= LibVarValue("phys_maxbarrier", "33");
-	aassettings.phys_jumpvel				= LibVarValue("phys_jumpvel", "270");
+	aassettings.phys_maxwaterjump			= LibVarValue("phys_maxwaterjump", "20");
+	aassettings.phys_maxbarrier				= LibVarValue("phys_maxbarrier", "43");
+	aassettings.phys_jumpvel				= LibVarValue("phys_jumpvel", "200");
 	aassettings.phys_falldelta5				= LibVarValue("phys_falldelta5", "40");
 	aassettings.phys_falldelta10			= LibVarValue("phys_falldelta10", "60");
 	aassettings.rs_waterjump				= LibVarValue("rs_waterjump", "400");
@@ -118,7 +118,7 @@
 	aassettings.rs_startelevator			= LibVarValue("rs_startelevator", "50");
 	aassettings.rs_falldamage5				= LibVarValue("rs_falldamage5", "300");
 	aassettings.rs_falldamage10				= LibVarValue("rs_falldamage10", "500");
-	aassettings.rs_maxfallheight			= LibVarValue("rs_maxfallheight", "0");
+	aassettings.rs_maxfallheight			= LibVarValue("rs_maxfallheight", "512");
 	aassettings.rs_maxjumpfallheight		= LibVarValue("rs_maxjumpfallheight", "450");
 } //end of the function AAS_InitSettings
 //===========================================================================
@@ -953,7 +953,7 @@
 			if (!gaptrace.startsolid)
 			{
 				//if it is a gap (lower than one step height)
-				if (gaptrace.endpos[2] < org[2] - aassettings.phys_maxstep - 1)
+				if (gaptrace.endpos[2] < org[2] - aassettings.phys_maxstep - 1) // Tobias CHECK: why phys_maxstep?
 				{
 					if (!(AAS_PointContents(end) & CONTENTS_WATER))
 					{
Index: code/botlib/be_ai_move.c
===================================================================
--- code/botlib/be_ai_move.c	(revision 3347)
+++ code/botlib/be_ai_move.c	(working copy)
@@ -3534,8 +3534,8 @@
 int BotSetupMoveAI(void)
 {
 	BotSetBrushModelTypes();
-	sv_maxstep = LibVar("sv_step", "18");
-	sv_maxbarrier = LibVar("sv_maxbarrier", "32");
+	sv_maxstep = LibVar("sv_maxstep", "19");
+	sv_maxbarrier = LibVar("sv_maxbarrier", "43");
 	sv_gravity = LibVar("sv_gravity", "800");
 	weapindex_rocketlauncher = LibVar("weapindex_rocketlauncher", "5");
 	weapindex_bfg10k = LibVar("weapindex_bfg10k", "9");
Index: code/botlib/be_ai_weap.c
===================================================================
--- code/botlib/be_ai_weap.c	(revision 3347)
+++ code/botlib/be_ai_weap.c	(working copy)
@@ -205,19 +205,19 @@
 	weaponconfig_t *wc;
 	weaponinfo_t weaponinfo;
 
-	max_weaponinfo = (int) LibVarValue("max_weaponinfo", "32");
+	max_weaponinfo = (int) LibVarValue("max_weaponinfo", "64");
 	if (max_weaponinfo < 0)
 	{
 		botimport.Print(PRT_ERROR, "max_weaponinfo = %d\n", max_weaponinfo);
-		max_weaponinfo = 32;
-		LibVarSet("max_weaponinfo", "32");
+		max_weaponinfo = 64;
+		LibVarSet("max_weaponinfo", "64");
 	} //end if
-	max_projectileinfo = (int) LibVarValue("max_projectileinfo", "32");
+	max_projectileinfo = (int) LibVarValue("max_projectileinfo", "64");
 	if (max_projectileinfo < 0)
 	{
 		botimport.Print(PRT_ERROR, "max_projectileinfo = %d\n", max_projectileinfo);
-		max_projectileinfo = 32;
-		LibVarSet("max_projectileinfo", "32");
+		max_projectileinfo = 64;
+		LibVarSet("max_projectileinfo", "64");
 	} //end if
 	Q_strncpyz(path, filename, sizeof(path));
 	PC_SetBaseFolder(BOTFILESBASEFOLDER);
Index: code/botlib/botlib.h
===================================================================
--- code/botlib/botlib.h	(revision 3347)
+++ code/botlib/botlib.h	(working copy)
@@ -427,18 +427,18 @@
 "phys_gravity"				"800"				be_aas_move.c		gravity value
 "phys_waterfriction"		"1"					be_aas_move.c		water friction
 "phys_watergravity"			"400"				be_aas_move.c		gravity in water
-"phys_maxvelocity"			"320"				be_aas_move.c		maximum velocity
-"phys_maxwalkvelocity"		"320"				be_aas_move.c		maximum walk velocity
+"phys_maxvelocity"			"260"				be_aas_move.c		maximum velocity
+"phys_maxwalkvelocity"		"280"				be_aas_move.c		maximum walk velocity
 "phys_maxcrouchvelocity"	"100"				be_aas_move.c		maximum crouch velocity
 "phys_maxswimvelocity"		"150"				be_aas_move.c		maximum swim velocity
 "phys_walkaccelerate"		"10"				be_aas_move.c		walk acceleration
 "phys_airaccelerate"		"1"					be_aas_move.c		air acceleration
 "phys_swimaccelerate"		"4"					be_aas_move.c		swim acceleration
-"phys_maxstep"				"18"				be_aas_move.c		maximum step height
+"phys_maxstep"				"19"				be_aas_move.c		maximum step height
 "phys_maxsteepness"			"0.7"				be_aas_move.c		maximum floor steepness
-"phys_maxbarrier"			"32"				be_aas_move.c		maximum barrier height
-"phys_maxwaterjump"			"19"				be_aas_move.c		maximum waterjump height
-"phys_jumpvel"				"270"				be_aas_move.c		jump z velocity
+"phys_maxbarrier"			"43"				be_aas_move.c		maximum barrier height
+"phys_maxwaterjump"			"20"				be_aas_move.c		maximum waterjump height
+"phys_jumpvel"				"200"				be_aas_move.c		jump z velocity
 "phys_falldelta5"			"40"				be_aas_move.c
 "phys_falldelta10"			"60"				be_aas_move.c
 "rs_waterjump"				"400"				be_aas_move.c
@@ -484,8 +484,8 @@
 "matchfile"					"match.c"			be_ai_chat.c		file with match strings
 "nochat"					"0"					be_ai_chat.c		disable chats
 "max_messages"				"1024"				be_ai_chat.c		console message heap size
-"max_weaponinfo"			"32"				be_ai_weap.c		maximum number of weapon info
-"max_projectileinfo"		"32"				be_ai_weap.c		maximum number of projectile info
+"max_weaponinfo"			"64"				be_ai_weap.c		maximum number of weapon info
+"max_projectileinfo"		"64"				be_ai_weap.c		maximum number of projectile info
 "max_iteminfo"				"256"				be_ai_goal.c		maximum number of item info
 "max_levelitems"			"256"				be_ai_goal.c		maximum number of level items
 
