Index: branches/rebase/code/botlib/be_ai_move.c
===================================================================
--- branches/rebase/code/botlib/be_ai_move.c	(revision 3373)
+++ branches/rebase/code/botlib/be_ai_move.c	(working copy)
@@ -1730,7 +1730,6 @@
 int BotAirControl(vec3_t origin, vec3_t velocity, vec3_t goal, vec3_t dir, float *speed)
 {
 	vec3_t org, vel;
-	float dist;
 	int i;
 
 	VectorCopy(origin, org);
@@ -1744,9 +1743,7 @@
 			VectorScale(vel, (goal[2] - org[2]) / vel[2], vel);
 			VectorAdd(org, vel, org);
 			VectorSubtract(goal, org, dir);
-			dist = VectorNormalize(dir);
-			if (dist > 32) dist = 32;
-			*speed = 400 - (400 - 13 * dist);
+			*speed = 400;
 			return qtrue;
 		} //end if
 		else
