Index: code/game/bg_pmove.c
===================================================================
--- code/game/bg_pmove.c	(revision 3352)
+++ code/game/bg_pmove.c	(working copy)
@@ -305,6 +305,34 @@
 	total = sqrt( cmd->forwardmove * cmd->forwardmove
 		+ cmd->rightmove * cmd->rightmove + cmd->upmove * cmd->upmove );
 	scale = (float)pm->ps->speed * max / ( 127.0 * total );
+	// ignore if in air
+	if (pm->ps->groundEntityNum == ENTITYNUM_NONE) {
+		return scale;
+	}
+	// ignore if spectator
+	if (pm->ps->persistant[PERS_TEAM] == TEAM_SPECTATOR) {
+		return scale;
+	}
+	// apply speed scale for strafing and going backwards
+	if (cmd->forwardmove < 0) {
+		scale *= 0.75f;
+	} else if (cmd->rightmove) {
+		scale *= 0.9f;
+	}
+	// running
+	if (!(pm->cmd.buttons & BUTTON_WALKING)) {
+		// apply weapon speed scale
+		switch (pm->ps->weapon) {
+			case WP_GAUNTLET:
+				scale *= 1.15f;
+				break;
+			case WP_RAILGUN:
+			case WP_BFG:
+				scale *= 0.9f;
+				break;
+			default:
+				break;
+		}
+	// walking
+	} else {
+		scale *= 0.75f;
+	}
 
 	return scale;
 }
@@ -1423,9 +1451,9 @@
 	
 
 	footstep = qfalse;
 
+	// ducked
 	if ( pm->ps->pm_flags & PMF_DUCKED ) {
-		bobmove = 0.5;	// ducked characters bob much faster
+		bobmove = 0.35f;
 		if ( pm->ps->pm_flags & PMF_BACKWARDS_RUN ) {
 			PM_ContinueLegsAnim( LEGS_BACKCR );
 		}
@@ -1432,35 +1460,29 @@
 		else {
 			PM_ContinueLegsAnim( LEGS_WALKCR );
 		}
-		// ducked characters never play footsteps
-	/*
+	// backwards
 	} else 	if ( pm->ps->pm_flags & PMF_BACKWARDS_RUN ) {
+		// running
 		if ( !( pm->cmd.buttons & BUTTON_WALKING ) ) {
-			bobmove = 0.4;	// faster speeds bob faster
+			bobmove = 0.57f;
 			footstep = qtrue;
+			PM_ContinueLegsAnim( LEGS_BACK );
+		// walking
 		} else {
-			bobmove = 0.3;
+			bobmove = 0.37f; // walking bobs slow
+			PM_ContinueLegsAnim(LEGS_BACKWALK);
 		}
-		PM_ContinueLegsAnim( LEGS_BACK );
-	*/
+	// forward
 	} else {
+		// running
 		if ( !( pm->cmd.buttons & BUTTON_WALKING ) ) {
-			bobmove = 0.4f;	// faster speeds bob faster
-			if ( pm->ps->pm_flags & PMF_BACKWARDS_RUN ) {
-				PM_ContinueLegsAnim( LEGS_BACK );
-			}
-			else {
-				PM_ContinueLegsAnim( LEGS_RUN );
-			}
+			bobmove = 0.47f;	// faster speeds bob faster
+			PM_ContinueLegsAnim( LEGS_RUN );
 			footstep = qtrue;
+		// walking
 		} else {
-			bobmove = 0.3f;	// walking bobs slow
-			if ( pm->ps->pm_flags & PMF_BACKWARDS_RUN ) {
-				PM_ContinueLegsAnim( LEGS_BACKWALK );
-			}
-			else {
-				PM_ContinueLegsAnim( LEGS_WALK );
-			}
+			bobmove = 0.35f;	// walking bobs slow
+			PM_ContinueLegsAnim( LEGS_WALK );
 		}
 	}
 
