Index: code/botlib/be_ai_move.c
===================================================================
--- code/botlib/be_ai_move.c	(revision 3360)
+++ code/botlib/be_ai_move.c	(working copy)
@@ -1891,10 +1891,10 @@
 //*
 bot_moveresult_t BotTravel_Jump(bot_movestate_t *ms, aas_reachability_t *reach)
 {
-	vec3_t hordir, dir1, dir2, start, end, runstart;
+	vec3_t hordir, dir1, dir2, dir3, start, end, runstart;
 //	vec3_t runstart, dir1, dir2, hordir;
 	int gapdist;
-	float dist1, dist2, speed;
+	float dist1, dist2, dist3, speed;
 	bot_moveresult_t_cleared( result );
 
+	speed = 400;
 	//
 	trap_AAS_JumpReachRunStart(reach, runstart, CONTENTS_SOLID|CONTENTS_PLAYERCLIP);
 	//*
@@ -1923,8 +1923,11 @@
 	VectorSubtract(ms->origin, runstart, dir2);
 	dir2[2] = 0;
 	dist2 = VectorNormalize(dir2);
+	VectorSubtract(runstart, reach->start, dir3);
+	dir3[2] = 0;
+	dist3 = VectorNormalize(dir3);
 	//if just before the reachability start
-	if (DotProduct(dir1, dir2) < -0.8 || dist2 < 5)
+	if (dist1 < dist3 + 10 || dist1 + 10 >= dist2 + dist3 || DotProduct(dir1, dir2) < -0.8 || dist2 < 5) // Tobias NOTE: why did I do: dist3 + 10? Still needed because otherwise we have a -dist2?
 	{
 //		botimport.Print(PRT_MESSAGE, "between jump start and run start point\n");
 		hordir[0] = reach->end[0] - ms->origin[0];
@@ -1975,8 +1975,8 @@
 		//elemantary action jump
-		if (dist1 < 24) EA_Jump(ms->client);
+		if (dist1 < 22) EA_Jump(ms->client); // 20 (for Railgun)
 		else if (dist1 < 32) EA_DelayedJump(ms->client);
-		EA_Move(ms->client, hordir, 600);
+		if (ms->moveflags & MFL_WALK && dist2 > dist3) {
+			speed = 200;
+		} else {
+			speed = 600;
+		}
+
+		EA_Move(ms->client, hordir, speed);
 		//
 		ms->jumpreach = ms->lastreachnum;
@@ -2002,11 +2002,11 @@
 		hordir[2] = 0;
 		VectorNormalize(hordir);
 		//
-		if (dist2 > 80) dist2 = 80;
-		speed = 400 - (400 - 5 * dist2);
+		if (ms->moveflags & MFL_WALK && dist2 > dist3) {
+			speed = 200;
+		} else {
+			if (dist2 > 100) {
+				dist2 = 100;
+			}
+
+			speed = 400 - (300 - 3 * dist2);
+		}
 		EA_Move(ms->client, hordir, speed);
 	} //end else
 	VectorCopy(hordir, result.movedir);