Index: code/botlib/be_ai_move.c
===================================================================
--- code/botlib/be_ai_move.c	(revision 3360)
+++ code/botlib/be_ai_move.c	(working copy)
@@ -1895,6 +1895,9 @@
 //	vec3_t runstart, dir1, dir2, hordir;
 	int gapdist;
 	float dist1, dist2, speed;
+#ifdef DEBUG
+	float currentspeed;
+#endif //DEBUG
 	bot_moveresult_t_cleared( result );
 
 	//
@@ -1926,11 +1929,14 @@
 	//if just before the reachability start
 	if (DotProduct(dir1, dir2) < -0.8 || dist2 < 5)
 	{
-//		botimport.Print(PRT_MESSAGE, "between jump start and run start point\n");
 		hordir[0] = reach->end[0] - ms->origin[0];
 		hordir[1] = reach->end[1] - ms->origin[1];
 		hordir[2] = 0;
 		VectorNormalize(hordir);
+#ifdef DEBUG
+		currentspeed = DotProduct(ms->velocity, hordir);
+		if (DotProduct(dir1, dir2) < -0.8) {
+			botimport.Print(PRT_MESSAGE, S_COLOR_CYAN "Between RE and RU: dist1 O to RE = %f, dist2 O to RU = %f, dist3 RU to RE = %f (%f, %f)\n", dist1, dist2, dist3, currentspeed, DotProduct(dir1, dir2));
+		} else {
+			botimport.Print(PRT_MESSAGE, S_COLOR_GREEN "Between RE and RU: dist1 O to RE = %f, dist2 O to RU = %f, dist3 RU to RE = %f (%f, %f)\n", dist1, dist2, dist3, currentspeed, DotProduct(dir1, dir2));
+		}
+#endif //DEBUG
 		//elemantary action jump
 		if (dist1 < 24) EA_Jump(ms->client);
+#ifdef DEBUG
+			botimport.Print(PRT_MESSAGE, S_COLOR_RED "Jumped! dist1 = %f (%f)\n", dist1, currentspeed);
+#endif //DEBUG
 		else if (dist1 < 32) EA_DelayedJump(ms->client);
+#ifdef DEBUG
+			botimport.Print(PRT_MESSAGE, S_COLOR_MAGENTA "Jump Delayed! dist1 = %f (%f)\n", dist1, currentspeed);
+#endif //DEBUG
@@ -1940,11 +1946,14 @@
 	} //end if
 	else
 	{
-//		botimport.Print(PRT_MESSAGE, "going towards run start point\n");
 		hordir[0] = runstart[0] - ms->origin[0];
 		hordir[1] = runstart[1] - ms->origin[1];
 		hordir[2] = 0;
 		VectorNormalize(hordir);
+#ifdef DEBUG
+		currentspeed = DotProduct(ms->velocity, hordir);
+		if (DotProduct(dir1, dir2) < -0.8) {
+			botimport.Print(PRT_MESSAGE, S_COLOR_BLUE "Going towards RU: dist1 O to RE = %f, dist2 O to RU = %f, dist3 RU to RE = %f (%f, %f)\n", dist1, dist2, dist3, currentspeed, DotProduct(dir1, dir2));
+		} else {
+			botimport.Print(PRT_MESSAGE, S_COLOR_YELLOW "Going towards RU: dist1 O to RE = %f, dist2 O to RU = %f, dist3 RU to RE = %f (%f, %f)\n", dist1, dist2, dist3, currentspeed, DotProduct(dir1, dir2));
+		}
+#endif //DEBUG
 		//
 		if (dist2 > 80) dist2 = 80;
 		speed = 400 - (400 - 5 * dist2);
