Index: code/botlib/be_ai_move.c
===================================================================
--- code/botlib/be_ai_move.c	(revision 3360)
+++ code/botlib/be_ai_move.c	(working copy)
@@ -1891,9 +1891,9 @@
 //*
 bot_moveresult_t BotTravel_Jump(bot_movestate_t *ms, aas_reachability_t *reach)
 {
-	vec3_t hordir, dir1, dir2, start, end, runstart;
+	vec3_t hordir, dir1, dir2, /*start, end, */runstart;
 //	vec3_t runstart, dir1, dir2, hordir;
-	int gapdist;
+//	int gapdist;
 	float dist1, dist2, speed;
 	bot_moveresult_t_cleared( result );
 
@@ -1900,6 +1900,7 @@
 	//
 	AAS_JumpReachRunStart(reach, runstart);
 	//*
+/* // Tobias NOTE: I'm pretty sure there will be maps where this piece of code would make sense, anyways, this code is NOT correct and it even causes bots to NOT jump although they should (e.g.: q3dm6) (FIXME?)
 	hordir[0] = runstart[0] - reach->start[0];
 	hordir[1] = runstart[1] - reach->start[1];
 	hordir[2] = 0;
@@ -1917,6 +1918,7 @@
 	} //end for
 	if (gapdist < 80) VectorMA(reach->start, gapdist, hordir, runstart);
 	//
+*/
 	VectorSubtract(ms->origin, reach->start, dir1);
 	dir1[2] = 0;
 	dist1 = VectorNormalize(dir1);
