Index: code/botlib/be_aas_reach.c
===================================================================
--- code/botlib/be_aas_reach.c	(revision 3374)
+++ code/botlib/be_aas_reach.c	(working copy)
@@ -4215,7 +4215,7 @@
 						//
 						VectorAdd(v1, v2, mid);
 						VectorScale(mid, 0.5, mid);
-						VectorMA(mid, 8, dir, mid);
+						VectorMA(mid, 31, dir, mid);
 						//
 						VectorCopy(mid, testend);
 						testend[2] -= 1000;
