Index: code/botlib/be_aas_move.c
===================================================================
--- code/botlib/be_aas_move.c	(revision 3379)
+++ code/botlib/be_aas_move.c	(working copy)
@@ -92,7 +92,7 @@
 	aassettings.phys_maxvelocity			= LibVarValue("phys_maxvelocity", "260");
 	aassettings.phys_maxwalkvelocity		= LibVarValue("phys_maxwalkvelocity", "280");
 	aassettings.phys_maxcrouchvelocity		= LibVarValue("phys_maxcrouchvelocity", "100");
-	aassettings.phys_maxswimvelocity		= LibVarValue("phys_maxswimvelocity", "150");
+	aassettings.phys_maxswimvelocity		= LibVarValue("phys_maxswimvelocity", "45");
 	aassettings.phys_walkaccelerate			= LibVarValue("phys_walkaccelerate", "10");
 	aassettings.phys_airaccelerate			= LibVarValue("phys_airaccelerate", "1");
 	aassettings.phys_swimaccelerate			= LibVarValue("phys_swimaccelerate", "4");
Index: code/botlib/botlib.h
===================================================================
--- code/botlib/botlib.h	(revision 3379)
+++ code/botlib/botlib.h	(working copy)
@@ -430,7 +430,7 @@
 "phys_maxvelocity"			"260"				be_aas_move.c		maximum velocity
 "phys_maxwalkvelocity"		"280"				be_aas_move.c		maximum walk velocity
 "phys_maxcrouchvelocity"	"100"				be_aas_move.c		maximum crouch velocity
-"phys_maxswimvelocity"		"150"				be_aas_move.c		maximum swim velocity
+"phys_maxswimvelocity"		"45"				be_aas_move.c		maximum swim velocity
 "phys_walkaccelerate"		"10"				be_aas_move.c		walk acceleration
 "phys_airaccelerate"		"1"					be_aas_move.c		air acceleration
 "phys_swimaccelerate"		"4"					be_aas_move.c		swim acceleration
Index: code/game/bg_pmove.c
===================================================================
--- code/game/bg_pmove.c	(revision 3379)
+++ code/game/bg_pmove.c	(working copy)
@@ -33,7 +33,7 @@
 // movement parameters
 float	pm_stopspeed = 100.0f;
 float	pm_duckScale = 0.25f;
-float	pm_swimScale = 0.50f;
+float	pm_swimScale = 0.15f;
 
 float	pm_accelerate = 10.0f;
 float	pm_airaccelerate = 1.0f;
@@ -511,20 +511,16 @@
 		PM_WaterJumpMove();
 		return;
 	}
-#if 0
+	if (pm->waterlevel == 3) {
 	// jump = head for surface
 	if ( pm->cmd.upmove >= 10 ) {
 		if (pm->ps->velocity[2] > -300) {
-			if ( pm->watertype & CONTENTS_WATER ) {
-				pm->ps->velocity[2] = 100;
-			} else if ( pm->watertype & CONTENTS_SLIME ) {
-				pm->ps->velocity[2] = 80;
+			if ( pm->watertype & CONTENTS_LAVA ) {
+				pm->ps->velocity[2] = 10;
 			} else {
 				pm->ps->velocity[2] = 50;
 			}
 		}
 	}
-#endif
+	}
 	PM_Friction ();
 
 	scale = PM_CmdScale( &pm->cmd );
@@ -534,7 +530,12 @@
 	if ( !scale ) {
 		wishvel[0] = 0;
 		wishvel[1] = 0;
-		wishvel[2] = -60;		// sink towards bottom
+		// sink towards bottom
+		if ( pm->watertype & CONTENTS_LAVA ) {
+			wishvel[2] = -10;
+		} else {
+			wishvel[2] = -60;
+		}
 	} else {
 		for (i=0 ; i<3 ; i++)
 			wishvel[i] = scale * pml.forward[i]*pm->cmd.forwardmove + scale * pml.right[i]*pm->cmd.rightmove;
@@ -549,7 +550,6 @@
 		wishspeed = pm->ps->speed * pm_swimScale;
 	}
 
-	PM_Accelerate (wishdir, wishspeed, pm_wateraccelerate);
 
+	if ( pm->watertype & CONTENTS_LAVA ) {
+		wishspeed *= 0.5;
+	}
 	// make sure we can go up slopes easily under water
 	if ( pml.groundPlane && DotProduct( pm->ps->velocity, pml.groundTrace.plane.normal ) < 0 ) {
@@ -566,6 +566,7 @@
 	}
 
 	PM_SlideMove( qfalse );
+	PM_Accelerate (wishdir, wishspeed, pm_wateraccelerate);
 }
 
 #ifdef MISSIONPACK
Index: code/game/g_active.c
===================================================================
--- code/game/g_active.c	(revision 3379)
+++ code/game/g_active.c	(working copy)
@@ -101,7 +101,7 @@
 	int			waterlevel;
 
 	if ( ent->client->noclip ) {
-		ent->client->airOutTime = level.time + 12000;	// don't need air
+		ent->client->airOutTime = level.time + 30000;	// don't need air
 		return;
 	}
 
@@ -136,7 +136,7 @@
 			}
 		}
 	} else {
-		ent->client->airOutTime = level.time + 12000;
+		ent->client->airOutTime = level.time + 30000;
 		ent->damage = 2;
 	}
 
Index: code/game/g_client.c
===================================================================
--- code/game/g_client.c	(revision 3379)
+++ code/game/g_client.c	(working copy)
@@ -1087,7 +1087,7 @@
 	client->ps.persistant[PERS_SPAWN_COUNT]++;
 	client->ps.persistant[PERS_TEAM] = client->sess.sessionTeam;
 
-	client->airOutTime = level.time + 12000;
+	client->airOutTime = level.time + 30000;
 
 	trap_GetUserinfo( index, userinfo, sizeof(userinfo) );
 	// set max health
