Index: code/ui/ui_local.h
===================================================================
--- code/ui/ui_local.h	(revision 3845)
+++ code/ui/ui_local.h	(working copy)
@@ -214,7 +214,7 @@
 #define MAX_GAMETYPES 16
 #define MAX_MAPS 128
 #define MAX_SPMAPS 16
-#define PLAYERS_PER_TEAM 5
+#define PLAYERS_PER_TEAM 7
 #define MAX_PINGREQUESTS		32
 #define MAX_ADDRESSLENGTH		64
 #define MAX_HOSTNAMELENGTH		22
Index: code/ui/ui_main.c
===================================================================
--- code/ui/ui_main.c	(revision 3845)
+++ code/ui/ui_main.c	(working copy)
@@ -1361,6 +1361,8 @@
 		case UI_BLUETEAM3:
 		case UI_BLUETEAM4:
 		case UI_BLUETEAM5:
+		case UI_BLUETEAM6:
+		case UI_BLUETEAM7:
 			value = trap_Cvar_VariableValue(va("ui_blueteam%i", ownerDraw-UI_BLUETEAM1 + 1));
 			if (value <= 0) {
 				text = "Closed";
@@ -1380,6 +1382,8 @@
 		case UI_REDTEAM3:
 		case UI_REDTEAM4:
 		case UI_REDTEAM5:
+		case UI_REDTEAM6:
+		case UI_REDTEAM7:
 			value = trap_Cvar_VariableValue(va("ui_redteam%i", ownerDraw-UI_REDTEAM1 + 1));
 			if (value <= 0) {
 				text = "Closed";
@@ -1698,6 +1702,8 @@
 		case UI_BLUETEAM3:
 		case UI_BLUETEAM4:
 		case UI_BLUETEAM5:
+		case UI_BLUETEAM6:
+		case UI_BLUETEAM7:
       UI_DrawTeamMember(&rect, scale, color, qtrue, ownerDraw - UI_BLUETEAM1 + 1, textStyle);
       break;
     case UI_REDTEAM1:
@@ -1705,6 +1711,8 @@
 		case UI_REDTEAM3:
 		case UI_REDTEAM4:
 		case UI_REDTEAM5:
+		case UI_REDTEAM6:
+		case UI_REDTEAM7:
       UI_DrawTeamMember(&rect, scale, color, qfalse, ownerDraw - UI_REDTEAM1 + 1, textStyle);
       break;
 		case UI_NETSOURCE:
@@ -2328,6 +2336,8 @@
 		case UI_BLUETEAM3:
 		case UI_BLUETEAM4:
 		case UI_BLUETEAM5:
+		case UI_BLUETEAM6:
+		case UI_BLUETEAM7:
       UI_TeamMember_HandleKey(flags, special, key, qtrue, ownerDraw - UI_BLUETEAM1 + 1);
       break;
     case UI_REDTEAM1:
@@ -2335,6 +2345,8 @@
 		case UI_REDTEAM3:
 		case UI_REDTEAM4:
 		case UI_REDTEAM5:
+		case UI_REDTEAM6:
+		case UI_REDTEAM7:
       UI_TeamMember_HandleKey(flags, special, key, qfalse, ownerDraw - UI_REDTEAM1 + 1);
       break;
 		case UI_NETSOURCE:
@@ -5322,6 +5334,8 @@
 vmCvar_t	ui_redteam3;
 vmCvar_t	ui_redteam4;
 vmCvar_t	ui_redteam5;
+vmCvar_t	ui_redteam6;
+vmCvar_t	ui_redteam7;
 vmCvar_t	ui_blueteam;
 vmCvar_t	ui_blueteam1;
 vmCvar_t	ui_blueteam2;
@@ -5328,6 +5342,8 @@
 vmCvar_t	ui_blueteam3;
 vmCvar_t	ui_blueteam4;
 vmCvar_t	ui_blueteam5;
+vmCvar_t	ui_blueteam6;
+vmCvar_t	ui_blueteam7;
 vmCvar_t	ui_teamName;
 vmCvar_t	ui_dedicated;
 vmCvar_t	ui_gameType;
@@ -5413,11 +5429,15 @@
 	{ &ui_redteam3, "ui_redteam3", "0", CVAR_ARCHIVE },
 	{ &ui_redteam4, "ui_redteam4", "0", CVAR_ARCHIVE },
 	{ &ui_redteam5, "ui_redteam5", "0", CVAR_ARCHIVE },
+	{ &ui_redteam6, "ui_redteam6", "0", CVAR_ARCHIVE },
+	{ &ui_redteam7, "ui_redteam7", "0", CVAR_ARCHIVE },
 	{ &ui_blueteam1, "ui_blueteam1", "0", CVAR_ARCHIVE },
 	{ &ui_blueteam2, "ui_blueteam2", "0", CVAR_ARCHIVE },
 	{ &ui_blueteam3, "ui_blueteam3", "0", CVAR_ARCHIVE },
 	{ &ui_blueteam4, "ui_blueteam4", "0", CVAR_ARCHIVE },
 	{ &ui_blueteam5, "ui_blueteam5", "0", CVAR_ARCHIVE },
+	{ &ui_blueteam6, "ui_blueteam6", "0", CVAR_ARCHIVE },
+	{ &ui_blueteam7, "ui_blueteam7", "0", CVAR_ARCHIVE },
 	{ &ui_netSource, "ui_netSource", "1", CVAR_ARCHIVE },
 	{ &ui_menuFiles, "ui_menuFiles", "ui/menus.txt", CVAR_ARCHIVE },
 	{ &ui_currentTier, "ui_currentTier", "0", CVAR_ARCHIVE },
Index: ui/menudef.h
===================================================================
--- ui/menudef.h	(revision 3845)
+++ ui/menudef.h	(working copy)
@@ -229,48 +229,52 @@
 #define UI_BLUETEAM3 212
 #define UI_BLUETEAM4 213
 #define UI_BLUETEAM5 214
-#define UI_REDTEAM1 215
-#define UI_REDTEAM2 216
-#define UI_REDTEAM3 217
-#define UI_REDTEAM4 218
-#define UI_REDTEAM5 219
-#define UI_NETSOURCE 220
-#define UI_NETMAPPREVIEW 221
-#define UI_NETFILTER 222
-#define UI_TIER 223
-#define UI_OPPONENTMODEL 224
-#define UI_TIERMAP1 225
-#define UI_TIERMAP2 226
-#define UI_TIERMAP3 227
-#define UI_PLAYERLOGO 228
-#define UI_OPPONENTLOGO 229
-#define UI_PLAYERLOGO_METAL 230
-#define UI_OPPONENTLOGO_METAL 231
-#define UI_PLAYERLOGO_NAME 232
-#define UI_OPPONENTLOGO_NAME 233
-#define UI_TIER_MAPNAME 234
-#define UI_TIER_GAMETYPE 235
-#define UI_ALLMAPS_SELECTION 236
-#define UI_OPPONENT_NAME 237
-#define UI_VOTE_KICK 238
-#define UI_BOTNAME 239
-#define UI_BOTSKILL 240
-#define UI_REDBLUE 241
-#define UI_CROSSHAIR 242
-#define UI_SELECTEDPLAYER 243
-#define UI_MAPCINEMATIC 244
-#define UI_NETGAMETYPE 245
-#define UI_NETMAPCINEMATIC 246
-#define UI_SERVERREFRESHDATE 247
-#define UI_SERVERMOTD 248
-#define UI_GLINFO  249
-#define UI_KEYBINDSTATUS 250
-#define UI_CLANCINEMATIC 251
-#define UI_MAP_TIMETOBEAT 252
-#define UI_JOINGAMETYPE 253
-#define UI_PREVIEWCINEMATIC 254
-#define UI_STARTMAPCINEMATIC 255
-#define UI_MAPS_SELECTION 256
+#define UI_BLUETEAM6 215
+#define UI_BLUETEAM7 216
+#define UI_REDTEAM1 217
+#define UI_REDTEAM2 218
+#define UI_REDTEAM3 219
+#define UI_REDTEAM4 220
+#define UI_REDTEAM5 221
+#define UI_REDTEAM6 222
+#define UI_REDTEAM7 223
+#define UI_NETSOURCE 224
+#define UI_NETMAPPREVIEW 225
+#define UI_NETFILTER 226
+#define UI_TIER 227
+#define UI_OPPONENTMODEL 228
+#define UI_TIERMAP1 229
+#define UI_TIERMAP2 230
+#define UI_TIERMAP3 231
+#define UI_PLAYERLOGO 232
+#define UI_OPPONENTLOGO 233
+#define UI_PLAYERLOGO_METAL 234
+#define UI_OPPONENTLOGO_METAL 235
+#define UI_PLAYERLOGO_NAME 236
+#define UI_OPPONENTLOGO_NAME 237
+#define UI_TIER_MAPNAME 238
+#define UI_TIER_GAMETYPE 239
+#define UI_ALLMAPS_SELECTION 240
+#define UI_OPPONENT_NAME 241
+#define UI_VOTE_KICK 242
+#define UI_BOTNAME 243
+#define UI_BOTSKILL 244
+#define UI_REDBLUE 245
+#define UI_CROSSHAIR 246
+#define UI_SELECTEDPLAYER 247
+#define UI_MAPCINEMATIC 248
+#define UI_NETGAMETYPE 249
+#define UI_NETMAPCINEMATIC 250
+#define UI_SERVERREFRESHDATE 251
+#define UI_SERVERMOTD 252
+#define UI_GLINFO  253
+#define UI_KEYBINDSTATUS 254
+#define UI_CLANCINEMATIC 255
+#define UI_MAP_TIMETOBEAT 256
+#define UI_JOINGAMETYPE 257
+#define UI_PREVIEWCINEMATIC 258
+#define UI_STARTMAPCINEMATIC 259
+#define UI_MAPS_SELECTION 260
 
 #define VOICECHAT_GETFLAG			"getflag"				// command someone to get the flag
 #define VOICECHAT_OFFENSE			"offense"				// command someone to go on offense
