Index: code/game/bg_misc.c
===================================================================
--- code/game/bg_misc.c	(revision 3382)
+++ code/game/bg_misc.c	(working copy)
@@ -1080,10 +1080,6 @@
 
 	case IT_ARMOR:
 #ifdef MISSIONPACK
-		if( bg_itemlist[ps->stats[STAT_PERSISTANT_POWERUP]].giTag == PW_SCOUT ) {
-			return qfalse;
-		}
-
 		if( bg_itemlist[ps->stats[STAT_PERSISTANT_POWERUP]].giTag == PW_GUARD ) {
 			upperBound = ps->stats[STAT_MAX_HEALTH];
 		}
Index: code/game/bg_pmove.c
===================================================================
--- code/game/bg_pmove.c	(revision 3382)
+++ code/game/bg_pmove.c	(working copy)
@@ -410,6 +410,11 @@
 
 	pm->ps->groundEntityNum = ENTITYNUM_NONE;
 	pm->ps->velocity[2] = JUMP_VELOCITY;
+
+	if( bg_itemlist[pm->ps->stats[STAT_PERSISTANT_POWERUP]].giTag == PW_SCOUT ) {
+		pm->ps->velocity[2] *= 1.5;
+	}
+
 	PM_AddEvent( EV_JUMP );
 
 	if ( pm->cmd.forwardmove >= 0 ) {
@@ -1011,6 +1016,10 @@
 	if ( pm->waterlevel == 2 ) {
 		delta *= 0.85;
 	}
+	// the scout powerup also reduces falling damage
+	if( bg_itemlist[pm->ps->stats[STAT_PERSISTANT_POWERUP]].giTag == PW_SCOUT ) {
+		delta *= 0.9;
+	}
 
 	if ( delta < 1 ) {
 		return;
@@ -1785,10 +1794,6 @@
 	}
 
 #ifdef MISSIONPACK
-	if( bg_itemlist[pm->ps->stats[STAT_PERSISTANT_POWERUP]].giTag == PW_SCOUT ) {
-		addTime /= 1.5;
-	}
-	else
 	if( bg_itemlist[pm->ps->stats[STAT_PERSISTANT_POWERUP]].giTag == PW_AMMOREGEN ) {
 		addTime /= 1.3;
 	}
Index: code/game/g_items.c
===================================================================
--- code/game/g_items.c	(revision 3382)
+++ code/game/g_items.c	(working copy)
@@ -153,7 +153,6 @@
 			handicap = 100.0f;
 		}
 		other->client->pers.maxHealth = handicap;
-		other->client->ps.stats[STAT_ARMOR] = 0;
 		break;
 
 	case PW_DOUBLER:
