Index: code/game/bg_pmove.c
===================================================================
--- code/game/bg_pmove.c	(revision 3384)
+++ code/game/bg_pmove.c	(working copy)
@@ -1795,7 +1795,7 @@
 
 #ifdef MISSIONPACK
 	if( bg_itemlist[pm->ps->stats[STAT_PERSISTANT_POWERUP]].giTag == PW_AMMOREGEN ) {
-		addTime /= 1.3;
+		addTime /= 1.1;
 	}
 	else
 #endif
Index: code/game/g_active.c
===================================================================
--- code/game/g_active.c	(revision 3384)
+++ code/game/g_active.c	(working copy)
@@ -475,17 +475,17 @@
 		  w = weapList[i];
 
 		  switch(w) {
-			  case WP_MACHINEGUN: max = 50; inc = 4; t = 1000; break;
+			  case WP_MACHINEGUN: max = 100; inc = 5; t = 1000; break;
 			  case WP_SHOTGUN: max = 10; inc = 1; t = 1500; break;
 			  case WP_GRENADE_LAUNCHER: max = 10; inc = 1; t = 2000; break;
 			  case WP_ROCKET_LAUNCHER: max = 10; inc = 1; t = 1750; break;
-			  case WP_LIGHTNING: max = 50; inc = 5; t = 1500; break;
+			  case WP_LIGHTNING: max = 100; inc = 10; t = 750; break;
 			  case WP_RAILGUN: max = 10; inc = 1; t = 1750; break;
-			  case WP_PLASMAGUN: max = 50; inc = 5; t = 1500; break;
-			  case WP_BFG: max = 10; inc = 1; t = 4000; break;
+			  case WP_PLASMAGUN: max = 50; inc = 5; t = 750; break;
+			  case WP_BFG: max = 20; inc = 1; t = 750; break;
 			  case WP_NAILGUN: max = 10; inc = 1; t = 1250; break;
 			  case WP_PROX_LAUNCHER: max = 5; inc = 1; t = 2000; break;
-			  case WP_CHAINGUN: max = 100; inc = 5; t = 1000; break;
+			  case WP_CHAINGUN: max = 100; inc = 10; t = 500; break;
 			  default: max = 0; inc = 0; t = 1000; break;
 		  }
 		  client->ammoTimes[w] += msec;
