Index: code/game/bg_misc.c
===================================================================
--- code/game/bg_misc.c	(revision 3383)
+++ code/game/bg_misc.c	(working copy)
@@ -1099,7 +1099,6 @@
 		return qtrue;
 
 	case IT_HEALTH:
-		// small and mega healths will go over the max, otherwise
 		// don't pick up if already at max
 #ifdef MISSIONPACK
 		if( bg_itemlist[ps->stats[STAT_PERSISTANT_POWERUP]].giTag == PW_GUARD ) {
@@ -1106,12 +1105,9 @@
 		}
 		else
 #endif
-		if ( item->quantity == 5 || item->quantity == 100 ) {
 			if ( ps->stats[STAT_HEALTH] >= ps->stats[STAT_MAX_HEALTH] * 2 ) {
 				return qfalse;
 			}
-			return qtrue;
-		}
 
 		if ( ps->stats[STAT_HEALTH] >= ps->stats[STAT_MAX_HEALTH] ) {
 			return qfalse;
Index: code/game/g_items.c
===================================================================
--- code/game/g_items.c	(revision 3383)
+++ code/game/g_items.c	(working copy)
@@ -278,7 +278,6 @@
 	int			max;
 	int			quantity;
 
-	// small and mega healths will go over the max
 #ifdef MISSIONPACK
 	if( bg_itemlist[other->client->ps.stats[STAT_PERSISTANT_POWERUP]].giTag == PW_GUARD ) {
 		max = other->client->ps.stats[STAT_MAX_HEALTH];
@@ -285,11 +284,7 @@
 	}
 	else
 #endif
-	if ( ent->item->quantity != 5 && ent->item->quantity != 100 ) {
-		max = other->client->ps.stats[STAT_MAX_HEALTH];
-	} else {
 		max = other->client->ps.stats[STAT_MAX_HEALTH] * 2;
-	}
 
 	if ( ent->count ) {
 		quantity = ent->count;
